dev/dev_requirements.txt

```diff
@@ -2,7 +2,14 @@ jupyter-book
 livereload
 black
 pytest
-mlflow-skinny[databricks]
```

Comment:
I don't think this is needed?  at least in the openai SDK folder, I was using poetry to install these locally and had mirrored that to the requirements.txt files there...

Reply:
Good point, this is the combined requirements that are installed in CI - but it makes sense to actually mirror user-facing env setup steps in CI (i.e. cd into `openai_sdk_agent_app_sample_code` and install dependencies via `poetry`). I'm hoping we can still do that effectively without checking in `poetry.lock`, unless that buys us better reproducibility than installing python libs via `requirements.txt`